lib/addons/argocd/index.ts

```diff
@@ -1,26 +1,359 @@
+import { HelmChart, KubernetesManifest, ServiceAccount } from "@aws-cdk/aws-eks";
+import { ManagedPolicy } from "@aws-cdk/aws-iam";
+import { SecretsManager } from "aws-sdk";
+
 import { ClusterAddOn, ClusterInfo, ClusterPostDeploy } from "../../stacks/cluster-types";
 import { Team } from "../../teams";
+import * as yaml from 'yaml';
+import { btoa } from '../../utils/string-utils';
+import { Construct } from "@aws-cdk/core";
+
+import bcrypt = require('bcrypt');
+
+
+export interface ArgoApplicationRepository {
+    /**
+     * Expected to support helm style repo at the moment
+     */
+    repoUrl: string,
+
+    /** 
+     * Path within the repository 
+     */
+    path?: string,
+
+    /**
+     * Optional name for the bootstrap application
+     */
+    name?: string,
+
+    /**
+     * Secret from AWS Secrets Manager to import credentials to access the specified git repository.
+     * The secret must exist in the same region and account where the stack will run. 
+     */
+    credentialsSecretName?: string,
+
+    /**
+     * Depending on credentials type the arn should either point to an SSH key (plain text value)
+     * or a json file with username/password attributes.
+     * For TOKEN type per ArgoCD documentation (https://argoproj.github.io/argo-cd/user-guide/private-repositories/) 
+     * username can be any non-empty username and token value as password.
+     */
+    credentialsType?: "USERNAME" | "TOKEN" | "SSH"
+
+}
+
+/**
+ * Configuration options for ArgoCD add-on.
+ */
+export interface ArgoCDAddOnProps {
+
+    /**
+     * Namespace where add-on will be deployed. 
+     */
+    namespace?: string,
+
+    /**
+     * If provided, the addon will bootstrap the app or apps in the provided repository.
+     * In general, the repo is expected to have the app of apps, which can enable to bootstrap all workloads,
+     * after the infrastructure and team provisioning is complete. 
+     */
+    bootstrapRepo?: ArgoApplicationRepository,
```

Comment:
should we make this an array? What if we want to have one repo for configuration and another for applications?

Reply:
I think it is a good idea, although who knows how they could diverge, e.g. options for Argo vs Flux. At present wego defined their own structure for it, so a bit late for convergence. However, agree that it is a generic structure. Will address.

Reply:
It is an app of apps, so it enables all sorts of combinations. They can have a separate repo for applications, separate repo (or multiple) for plugins and just reference them in the app of apps. We could make it an array but it sort opens up non-GitOps onboarding, e.g. by modifying the code and adding another repo. I think we should force them to use app of apps properly.

Reply:
sort of...I thinking of an example where you have sep repos for applications and configuration.